### ISSUE TITLE
Unstable test case during validation

### ISSUE THREAD

**Issue by animationzl** [cloudfoundry-incubator/cf-openstack-validator#29]
- 'can save and retrieve user-data from metadata service'
- 'allows one VM to reach port 22 of another VM within the same network'

These two test cases pass occasionally depending on the `time_in_seconds` parameter passed to the `execute_ssh_command_on_vm_with_retry` function, for example according to my test against devstack environment, they will pass with the value of `time_in_seconds` set to 300, but fail with 60 which is the default value [1].

[1] https://github.com/cloudfoundry-incubator/cf-openstack-validator/blob/master/lib/validator/api/cpi_helpers.rb#L23

**Comment by voelzmo** [comment]
Hey @animationzl,

Thanks for reporting this. A quick question: how long does it take for a VM to boot on your DevStack? It might be that you're running DevStack on a slightly underpowered machine if it takes 5 minutes to be ready?

**Comment by voelzmo** [comment]
Hey @animationzl 

I'm not sure about your tempest reference here. Are you saying it can take more than 60 seconds for the TCP connection to build up? I'm not sure if it makes sense to run those tests then on the machine you're currently running them on. Maybe we just skip them?

**Comment by animationzl** [comment]
@voelzmo yes, it can take more than 60s for the TCP connection to build up, according to my test, the test cases will pass with `time_in_seconds` set to 200s, but fail with default 60s

**Comment by kiwik** [comment]
OpenLab build OpenStack deployment in a virtual machine provisioned by public cloud, so it might cause more ssh access time in validator tests, I think if we can add a `ssh_connect_timeout` option into validator.yml, that can address the special case.
